DatadogRUM/Sources/Instrumentation/AppHangs/AppHangsWatchdogThread.swift

```diff
@@ -41,6 +43,16 @@ internal final class AppHangsWatchdogThread: Thread {
     private let mainQueue: DispatchQueue
     /// SDK date provider.
     private let dateProvider: DateProvider
+    /// Backtrace reporter for hang's stack trace generation.
+    private let backtraceReporter: BacktraceReporting
+    /// An identifier of the main thread required for backtrace generation.
+    /// Because backtrace is generated from the watchdog thread, we must identify the main thread to be promoted in `BacktraceReport`. This value is
+    /// obtained at runtime, only once and it is cached for later use.
+    @ReadWriteLock
+    private var mainThreadID: ThreadID? = nil
+    /// If `mainThreadID` value was already obtained.
+    /// This flag doesn't require synchronization as it is only used from the main thread.
+    private var isMainThreadIDCached = false
```

Comment:
/blocking The `isMainThreadIDCached` is not necessary since the main queue is serial:
If we do the following during the init:
```swift
if Thread.isMainThread {
    // When initialization happens on the main thread, we can get its `ThreadID` right away, so startup hangs are covered
    self.mainThreadID = backtraceReporter.currentThreadID()
} else 
  mainQueue.async { [weak self] in
     self?.mainThreadID = backtraceReporter.currentThreadID()
  }
}
```
It will execute before the execution of `mainQueue.async {}` in the start method, no need to get the ID there.

Reply:
Ignore my previous comment, the feedback is all right as discussed on Slack 👍. I made the change ✅ - it greatly simplified the flow 👍, good call.